ambry-api/src/main/java/com/github/ambry/router/Router.java

```diff
@@ -34,10 +35,12 @@
    * @param blobId The ID of the blob for which blob data is requested.
    * @param options The options associated with the request. This cannot be null.
    * @param callback The callback which will be invoked on the completion of the request.
+   * @param quotaChargeEventListener Listener interface to charge quota cost for the operation.
    * @return A future that would eventually contain a {@link GetBlobResult} that can contain either
    *         the {@link BlobInfo}, the {@link ReadableStreamChannel} containing the blob data, or both.
    */
-  Future<GetBlobResult> getBlob(String blobId, GetBlobOptions options, Callback<GetBlobResult> callback);
+  Future<GetBlobResult> getBlob(String blobId, GetBlobOptions options, Callback<GetBlobResult> callback,
+      QuotaChargeEventListener quotaChargeEventListener);
```

Comment:
And can we rename this to something else, like `ChunkEventListener`. I think it can be a generic callback not just for quota charging logic.

Reply:
The reason why I wanted to keep it as quota specific, instead of a generic interface, is that its not necessary that quota charge will happen on chunks. For example, in case of tll, delete and undelete, we are doing quota charge on the whole operation, and not on individual chunks. So I pass the `QuotaChargeEventListener` to each type `Operation` (e.g, `GetOperation`, `DeleteOperation` etc), and let the operation class itself decide where the quota charge needs to happen (on completion of main operation, or for each chunk). Let me know what you think about this.

The other alternative is to do quota charging at different places for different type of Ambry requests. I feel that this option scatters the quota code everywhere in the stack and something uniform is better.